cmd/kube-apiserver/app/server.go

```diff
@@ -569,7 +575,43 @@ func BuildStorageFactory(s *options.ServerRunOptions) (*serverstorage.DefaultSto
 	}
 
 	if s.Etcd.EncryptionProviderConfigFilepath != "" {
-		transformerOverrides, err := encryptionconfig.GetTransformerOverrides(s.Etcd.EncryptionProviderConfigFilepath)
+		// Obtain GCE cloud for a Google Cloud KMS connection if needed.
+		// gce package cannot be accessed inside staging, which is why this function is passed from here.
+		getGCECloud := func() (*kms.GCECloud, error) {
+			cloud, err := cloudprovider.InitCloudProvider(s.CloudProvider.CloudProvider, s.CloudProvider.CloudConfigFile)
+			if err != nil {
+				return nil, err
+			}
+
+			var cloudkmsService *cloudkms.Service
+			var projectID string
+
+			// Safe when cloud is nil too.
+			if gke, ok := cloud.(*gce.GCECloud); ok {
+				// Hosting on GCE/GKE with Google KMS encryption provider
+				cloudkmsService = gke.GetKMSService()
+
+				// Project ID is assumed to be the user's project unless there
+				// is an override in the configuration file
+				projectID = gke.GetProjectID()
```

Comment:
[Here](https://github.com/sakshamsharma/kubernetes/blob/63d46ffebd2ca3bb9daf1d87c6c1c28ed7680a93/staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/kms/google.go#L44)

I will make the comment here clearer.